src/commands/dev/dev.mjs

```diff
@@ -55,16 +55,29 @@ import { createDevExecCommand } from './dev-exec.mjs'
 const netlifyBuildPromise = import('@netlify/build')
 
 const startStaticServer = async ({ settings }) => {
-  const server = new StaticServer({
-    rootPath: settings.dist,
-    name: 'netlify-dev',
-    port: settings.frameworkPort,
-    templates: {
-      notFound: path.join(settings.dist, '404.html'),
+  const server = Fastify()
+  server.register(fastifyStatic, {
+    root: path.resolve(settings.dist),
+    // eslint-disable-next-line no-unused-vars
+    setHeaders: (res, _path, _stat) => {
+      res.setHeader('X-Powered-by', 'netlify-dev')
```

Comment:
The `name: 'netlify-dev'` configuration option in `static-server` sets the `X-Powered-by` header.